src/Illuminate/Database/Connection.php

```diff
@@ -595,6 +595,8 @@ public function prepareBindings(array $bindings)
                 $bindings[$key] = $value->format($grammar->getDateFormat());
             } elseif ($value === false) {
                 $bindings[$key] = 0;
+            } elseif ($value === true) {
```

Comment:
How about this?

```php
             if ($value instanceof DateTimeInterface) {
                 $bindings[$key] = $value->format($grammar->getDateFormat());
             } elseif (is_bool($value)) {
                 $bindings[$key] = (int) $value;
             }
```

Shows the intent that we want to convert booleans to integers.

Reply:
Yeah this change would be good I think.